specs/lib/convertSpec.js

```diff
@@ -1,23 +1,160 @@
+                    smoothNormals : false,
+                    optimizeForCesium : false,
+                    textureCompressionOptions : undefined,
+                    preserve : true
+                });
+            }), done).toResolve();
+    });
+
+    it('sets options', function(done) {
+        var spy1 = spyOn(GltfPipeline, 'processJSONToDisk');
+        var spy2 = spyOn(writeUris, '_outputFile');
+        var textureCompressionOptions = {
+            format : 'dxt1',
+            quality : 10
+        };
+        var options = {
+            binary : true,
+            separate : true,
+            separateTextures : true,
+            compress : true,
+            optimize : true,
+            generateNormals : true,
+            ao : true,
+            kmc : true,
+            optimizeForCesium : true,
+            textureCompressionOptions : textureCompressionOptions
+        };
+
+        expect(convert(objPath, gltfPath, options)
+            .then(function() {
+                var args = spy1.calls.first().args;
+                var options = args[2];
+                expect(options).toEqual({
+                    createDirectory : false,
+                    basePath : path.dirname(objPath),
+                    binary : true,
+                    embed : false,
+                    embedImage : false,
+                    encodeNormals : true,
+                    quantize : true,
+                    compressTextureCoordinates : true,
+                    aoOptions : {},
+                    kmcOptions : {},
+                    smoothNormals : true,
+                    optimizeForCesium : true,
+                    textureCompressionOptions : textureCompressionOptions,
+                    preserve : false
+                });
+                expect(spy2.calls.count()).toBe(2); // Saves out .png and .bin
+            }), done).toResolve();
+    });
+
+    it('saves as binary if gltfPath has a .glb extension', function(done) {
+        var spy = spyOn(GltfPipeline, 'processJSONToDisk');
+        expect(convert(objPath, glbPath)
             .then(function() {
                 var args = spy.calls.first().args;
-                expect(args[0]).toBeDefined();
-                expect(path.normalize(args[1])).toEqual(path.normalize(gltfFile));
+                var options = args[2];
+                expect(options.binary).toBe(true);
             }), done).toResolve();
     });
+
+    it('bypassPipeline flag bypasses gltf-pipeline', function(done) {
+        var spy1 = spyOn(convert, '_outputJson');
+        var spy2 = spyOn(GltfPipeline, 'processJSONToDisk');
+        var options = {
+            bypassPipeline : true
+        };
+        expect(convert(objPath, gltfPath, options)
+            .then(function() {
+                expect(spy1.calls.count()).toBe(1);
+                expect(spy2.calls.count()).toBe(0);
+            }), done).toResolve();
+    });
+
+    it('uses a custom logger', function(done) {
```

Comment:
This test isn't really useful (and is also fragile if we change what we're logging).  We should be testing for specific log messages in each instance we expect one to be generated, not just running something and seeing if the logger gets called.